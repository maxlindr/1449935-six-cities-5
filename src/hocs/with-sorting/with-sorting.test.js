import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withSorting from './with-sorting';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withSorting(MockComponent);

test(`withSorting должен корректно отображаться`, () => {
  const tree = renderer.create((
    <MockComponentWrapped />
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
