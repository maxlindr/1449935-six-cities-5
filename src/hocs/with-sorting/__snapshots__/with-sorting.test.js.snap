// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSorting должен корректно отображаться 1`] = `
<div>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by 
    </span>
    <span
      className="places__sorting-type"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
    <ul
      className="places__options places__options--custom"
    >
      <li
        className="places__option places__option--active"
        data-value="popular"
        onClick={[Function]}
        tabIndex="0"
      >
        Popular
      </li>
      <li
        className="places__option"
        data-value="low-to-high"
        onClick={[Function]}
        tabIndex="0"
      >
        Price: low to high
      </li>
      <li
        className="places__option"
        data-value="high-to-low"
        onClick={[Function]}
        tabIndex="0"
      >
        Price: high to low
      </li>
      <li
        className="places__option"
        data-value="top-rated"
        onClick={[Function]}
        tabIndex="0"
      >
        Top rated first
      </li>
    </ul>
  </form>
</div>
`;
